client/test/root-section.ts

```diff
@@ -135,4 +139,32 @@ describe( 'Logged In Landing Page', () => {
 
 		await waitFor( () => expect( page.current ).toBe( '/sites' ) );
 	} );
+
+	test( 'user with a selected site and edit permissions goes to My Home', async () => {
+		const state = {
+			currentUser: {
+				id: 1,
+				capabilities: { 1: { edit_posts: true }, 2: { edit_posts: true } },
+				user: { primary_blog: 1 },
+			},
+			ui: { selectedSiteId: 2 },
+			sites: {
+				items: {
+					1: {
+						ID: 1,
+						URL: 'https://test.wordpress.com',
+					},
+					2: {
+						ID: 2,
+						URL: 'https://selected.wordpress.com',
+					},
+				},
+			},
+		};
+		const { page } = initRouter( { state } );
+
+		page( '/' );
+
+		await waitFor( () => expect( page.current ).toBe( '/home/selected.wordpress.com' ) );
+	} );
```

Comment:
It feels like we should also add a test case to make sure that users with WPAdmin as their preferred UI are redirected correctly too.